src/routes/events/events.controller.js

```diff
@@ -0,0 +1,70 @@
+const fs = require('fs');
+const { promisify } = require('util');
+const writeFile = promisify(fs.writeFile);
+const eventsData = require('../../../db/events.data.json');
+
+const listEvents = (req, res) => {
+  res.status(200);
+  return res.json({ data: eventsData });
+};
+
+const postNewEvent = async (req, res) => {
+  const id = eventsData.length + 1;
+  const newEventsData = [...eventsData, { id, ...req.body }];
+  await writeFile('db/events.data.json', JSON.stringify(newEventsData));
+  res.status(201);
+
+  return res.json({
+    id,
+    ...req.body
+  });
+};
+
+const updateEvent = async (req, res) => {
+  const id = req.params.id;
+  const events = eventsData;
+  const index = events.findIndex(event => event.id == id);
+
+  if (index === -1) {
+    return res.status(404).json({ errors: 'event id not found' });
+  }
+  const updatedEventsData = eventsData.map(event => {
+    if (event.id == id) {
+      return {
+        ...event,
+        ...req.body
+      };
+    }
+    return event;
+  });
+
+  await writeFile('db/events.data.json', JSON.stringify(updatedEventsData));
+  res.status(200);
+  return res.json({
+    updated: id
+  });
+};
+
+const deleteEvent = async (req, res) => {
+  const id = req.params.id;
+  const updatedEventsData = eventsData;
+  const index = updatedEventsData.findIndex(event => event.id == id);
+
+  if (index === -1) {
+    return res.status(404).json({ errors: 'event id not found' });
+  }
+
+  updatedEventsData.splice(index, 1);
+  await writeFile('db/events.data.json', JSON.stringify(updatedEventsData));
+  res.status(200);
```

Comment:
This isn't the right HTTP code for deletion. Can you find out what it is?

Reply:
I got the `200` from here https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE, do you think `204` is more appropriate or is it something else entirely!?

Reply:
Sorry, I overstated that a bit. 200 OK *is* a valid response code for a DELETE request, but in this case you aren't returning anything in the response body, which makes 200 OK a little inappropriate.  If you want to return 200 OK, you should include a representation of the deleted record in the response payload. I personally use 204 NO CONTENT for successful deletions, because it's accurately reporting the new status of the record at the requested URL - it's deleted, so there's no content to return.